clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7067,6 +7068,19 @@ void OffloadBundler::ConstructJob(Compilation &C, const JobAction &JA,
       Triples += CurDep->getOffloadingArch();
     }
   }
+  bool IsFPGADepBundle = (TCArgs.hasArg(options::OPT_fintelfpga) &&
+      Output.getType() == types::TY_Object);
+  // For -fintelfpga, when bundling objects we also want to bundle up the
+  // named dependency file.
+  // TODO - We are currently using the target triple inputs to slot a location
+  // of the dependency information into the bundle.  It would be good to
+  // separate this out to an explicit option in the bundler for the dependency
+  // file as it does not match the type being bundled.
+  if (IsFPGADepBundle) {
+    Triples += ',';
+    Triples += Action::GetOffloadKindName(Action::OFK_SYCL);;
+    Triples += "-spir64_fpga_depfile";
```

Comment:
I'd suggest 'computing' the magic string once for both usages (if `spir64_fpga` can be retrieved via `llvm::Triple`'s string-returning functions)

Reply:
Done - created an fpga_dep arch type to handle this.